diff --git a/arch/arm/boot/dts/qcom-ipq4018-gl-ap1300.dts b/arch/arm/boot/dts/qcom-ipq4018-gl-ap1300.dts
index 41ed66e595007..35b5a245e860b 100644
--- a/arch/arm/boot/dts/qcom-ipq4018-gl-ap1300.dts
+++ b/arch/arm/boot/dts/qcom-ipq4018-gl-ap1300.dts
@@ -14,7 +14,6 @@
 		led-failsafe = &led_power;
 		led-running = &led_power;
 		led-upgrade = &led_power;
-		label-mac-device = &gmac0;
 	};
 
 	memory {
@@ -187,6 +186,14 @@
 				precal_art_5000: precal@5000 {
 					reg = <0x5000 0x2f20>;
 				};
+
+				macaddr_art_e010: mac-address@e010 {
+					reg = <0xe010 0x6>;
+				};
+
+				macaddr_art_e018: mac-address@e018 {
+					reg = <0xe018 0x6>;
+				};
 			};
 		};
 	};
@@ -256,6 +263,30 @@
 	status = "okay";
 };
 
+&gmac {
+	status = "okay";
+};
+
+&switch {
+	status = "okay";
+};
+
+&swport4 {
+	status = "okay";
+	label = "lan";
+
+	nvmem-cells = <&macaddr_art_e018>;
+	nvmem-cell-names = "mac-address";
+};
+
+&swport5 {
+	status = "okay";
+	label = "wan";
+
+	nvmem-cells = <&macaddr_art_e010>;
+	nvmem-cell-names = "mac-address";
+};
+
 &wifi0 {
 	status = "okay";
 	nvmem-cell-names = "pre-calibration";

