diff --git a/arch/arm/boot/dts/qcom-ipq4018-gl-ap1300.dts b/arch/arm/boot/dts/qcom-ipq4018-gl-ap1300.dts
index fa81bfbf5..cb42a209b 100644
--- a/arch/arm/boot/dts/qcom-ipq4018-gl-ap1300.dts
+++ b/arch/arm/boot/dts/qcom-ipq4018-gl-ap1300.dts
@@ -22,7 +22,7 @@ memory {
 	};
 
 	chosen {
-		bootargs-append = " ubi.mtd=ubi root=/dev/ubiblock0_1";
+		bootargs-append = " ubi.mtd=ubi root=ubi0:rootfs rootfstype=ubifs";
 	};
 
 	soc {
